spark/src/main/java/org/apache/iceberg/spark/SparkUtil.java

```diff
@@ -38,6 +41,14 @@
   private SparkUtil() {
   }
 
+  public static Table toSerializedTable(Table table) {
```

Comment:
What about a solution like `HasTableOperations`?
My feeling is that the Serialization method should be the responsibility of the table itself.

What do you think?

Reply:
You mean like a `HasSerializableProxy` that exposes a `toSerializable()` method?

Reply:
How will we pass a custom FileIO required by serialization in Spark?